--- tuxwordsmith.py.orig	2010-01-15 14:36:13.000000000 +0100
+++ tuxwordsmith.py	2011-04-15 23:54:33.818546856 +0200
@@ -17,6 +17,7 @@
 ***********************************************************/
 """
 import os,sys,string,time
+sys.path.append('/usr/share/games/tuxwordsmith/lib')
 from TuxWordSmith.tws import *
 
 def usage():
--- TuxWordSmith/tws.py.orig	2010-10-26 15:05:43.000000000 +0200
+++ TuxWordSmith/tws.py	2011-04-17 18:07:55.200020894 +0200
@@ -1285,7 +1285,7 @@ class TuxWordSmith(TWSSolver):
 		infname=os.path.join(homedir,'.tws_config')
 		
 		if not os.path.exists(infname):
-			master_fname=os.path.join(self.env.fontdir,'.tws_config_master')
+			master_fname=os.path.join('/','usr','share','games','tuxwordsmith','.tws_config_master')
 			if self.env.OS=='win':
 				cmd="copy %s \"%s\""%(master_fname,os.path.join(homedir,'.tws_config'))
 				os.system(cmd)
--- TuxWordSmith/environment.py.orig	2010-10-26 14:28:10.000000000 +0200
+++ TuxWordSmith/environment.py	2011-04-17 18:13:59.960028122 +0200
@@ -35,10 +35,9 @@ class Environment:
 		
 		if OS=='lin':
 			import pygame 
-			for sitepkgdir in sys.path:
-				if sitepkgdir[-13:]=='site-packages':break
-			configdir=os.path.join('/','var','games',appname)#,os.path.basename(appname)
-			fontdir=os.path.join('/','var','games',appname)#,os.path.basename(appname)
+			sitepkgdir=os.path.join('/','usr','share','games','tuxwordsmith', 'lib')
+			configdir=os.path.join('/','usr','share','games','tuxwordsmith')#,os.path.basename(appname)
+			fontdir=os.path.join('/','usr','share','games','tuxwordsmith')#,os.path.basename(appname)
 			homedir=os.getenv('HOME')
 		
 		elif OS=='win':
@@ -57,25 +56,6 @@ class Environment:
 			fontdir='.'
 			homedir=os.getenv('HOME')
 		
-		
-		#If application hasn't been installed (via setup.py) then try to run from tgz directory:
-		if os.path.exists(appname) and os.path.exists('tuxwordsmith.py'):#if yes, then probably want to be running w/o installing.
-			sitepkgdir='.'
-			fontdir='.'
-			configdir='.'
-			homedir=os.getenv('HOME')
-			if not homedir:homedir=os.getenv('USERPROFILE')
-		elif os.path.exists(os.path.join(sitepkgdir,appname,'tws.py')):pass
-		elif  os.path.exists(os.path.join(sitepkgdir,appname,'tws.py')) and OS=='win':pass
-		else:
-			sitepkgdir='.'
-			fontdir='.'
-			configdir='.'
-			homedir=os.getenv('HOME')
-			if not homedir:homedir=os.getenv('USERPROFILE')
-			if not homedir:homedir='.'
-		
-			
 		self.OS=OS
 		self.sitepkgdir=sitepkgdir
 		self.fontdir=fontdir
